src/App.tsx

```diff
@@ -1,24 +1,88 @@
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useMemo, useState } from 'react';
+import { TodosList } from './components/TodosList';
+import { FilterBy, TodosFooter } from './components/TodosFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoHeader } from './components/TodoHeader';
+import { useDispatch, useSelector } from './providers/TodosContext';
+import { filterTodos } from './helpers/filterTodo';
+import { deleteTodo } from './api/todos';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const {
+    todos,
+    inProcess,
+    isError,
+    errorMessage,
+    updateTodos,
+  } = useSelector();
+  const dispatch = useDispatch();
+
+  const [filterBy, setFilterBy] = useState<FilterBy>('all');
+
+  const preparedTodos = useMemo(() => {
+    return filterTodos(todos, filterBy);
+  }, [filterBy, todos]);
+
+  const completedTodos = useMemo(() => {
+    return todos.filter(
+      ({ completed }) => completed,
+    );
+  }, [todos]);
+  const activeTodosLength = todos.length - completedTodos.length;
+  const isSomeActive = todos.some(({ completed }) => !completed);
+
+  const handleDeleteCompleted = () => {
+    const completedIds = completedTodos.map(({ id }) => (id));
+
+    dispatch({
+      type: 'setInProcess',
+      payload: [...inProcess, ...completedIds],
+    });
+    Promise.all(completedIds.map((id) => deleteTodo(id)))
+      .then(() => {
+        updateTodos();
+      })
+      .catch(() => {
+        dispatch({
+          type: 'setError',
+          payload: { isError: true, errorMessage: 'Unable to delete a todo' },
+        });
+      })
+      .finally(() => {
+        dispatch({
+          type: 'setInProcess',
+          payload: [],
+        });
+      });
+  };
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoHeader isSomeActive={isSomeActive} />
+
+        <TodosList todos={preparedTodos} />
+
+        {todos.length > 0 && (
+          <TodosFooter
+            filterBy={filterBy}
+            activeTodosLength={activeTodosLength}
+            completedTodosLength={completedTodos.length}
+            onFilterChange={setFilterBy}
+            onDeleteCompleted={handleDeleteCompleted}
+          />
+        )}
+      </div>
+
+      <ErrorNotification
+        isHidden={!isError}
+        message={errorMessage}
+        onClose={
+          () => dispatch({ type: 'setError', payload: { isError: false } })
```

Comment:
Pls move the function inside `onClose` handler to a separate const and use as a callback here